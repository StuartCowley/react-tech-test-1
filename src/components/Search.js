import React, { useState } from "react";
import PropTypes from "prop-types";
import "../styles/Search.css";
import getImages from "../requests/getImages";

const Search = ({ setSearchResults, setUserHasSearched }) => {
  const [value, setValue] = useState();

  const handleSubmit = async (event) => {
    event.preventDefault();
    setSearchResults(await getImages(value));
    setUserHasSearched(true);
  };

  return (
    <div className="search">
      <form className="search-form" onSubmit={handleSubmit}>
        <input
          className="search-form__input"
          type="text"
          onChange={(event) => setValue(event.target.value)}
        />
        <button className="search-form__button" type="submit">
          Search
        </button>
      </form>
    </div>
  );
};

Search.propTypes = {
  setSearchResults: PropTypes.func.isRequired,
  setUserHasSearched: PropTypes.func.isRequired,
};

export default Search;
