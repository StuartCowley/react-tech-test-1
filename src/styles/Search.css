.search-form {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.search-form__input {
  width: 20%;
  height: 50px;
  border-radius: 11px;
  font-size: 20px;
  padding: 0 10px;
}

.search-form__button {
  margin: 10px 0 0 10px;
  height: 35px;
  width: 80px;
  border-radius: 11px;
  font-size: 15px;
  background-color: lightgray;
}

.search-form__button:hover {
  background-color: rgb(107, 107, 107);
  color: white;
}
