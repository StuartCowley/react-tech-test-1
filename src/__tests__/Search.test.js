import React from "react";
import Search from "../components/Search";
import { render } from "@testing-library/react";

describe("Search", () => {
  const { asFragment } = render(
    <Search setSearchResults={jest.fn()} setUserHasSearched={jest.fn()} />
  );

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
