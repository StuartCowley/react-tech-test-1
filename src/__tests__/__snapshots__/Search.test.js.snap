// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <form
      class="search-form"
    >
      <input
        class="search-form__input"
        type="text"
      />
      <button
        class="search-form__button"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
